extensions/WSLExtension/Models/WslComputeSystem.cs

```diff
@@ -0,0 +1,346 @@
+        get
+        {
+            if (GetState() == ComputeSystemState.Stopped)
+            {
+                return ComputeSystemOperations.Delete;
+            }
+
+            return ComputeSystemOperations.Delete | ComputeSystemOperations.Terminate;
+        }
+    }
+
+    public WslComputeSystem(
+        IStringResource stringResource,
+        WslRegisteredDistribution distribution,
+        IWslManager wslManager)
+    {
+        _stringResource = stringResource;
+        _distribution = distribution;
+        _wslManager = wslManager;
+        DisplayName = distribution.FriendlyName;
+        _curPackage = _packageHelper.GetPackageFromPackageFamilyName(_distribution.PackageFamilyName!);
+
+        // Use display name of package if there is no friendly name for this distribution
+        if (string.IsNullOrEmpty(DisplayName) && _curPackage != null)
+        {
+            DisplayName = _curPackage.DisplayName;
+        }
+
+        // Use the unique name of the distribution as its supplemental name as long as its not the same
+        // as the display name above.
+        if (!DisplayName.Equals(distribution.Name, StringComparison.OrdinalIgnoreCase))
+        {
+            SupplementalDisplayName = distribution.Name;
+        }
+
+        Id = distribution.Name;
+        _wslManager.DistributionStateSyncEventHandler += OnStateSyncRequested;
+        _versionLabel = _stringResource.GetLocalized("WSLVersionLabel");
+        _defaultDistributionLabel = _stringResource.GetLocalized("WSLDefaultDistributionLabel");
+        _defaultDistributionValue = _stringResource.GetLocalized("WSLDefaultDistributionValue");
+        _publisherLabel = _stringResource.GetLocalized("WSLPublisherLabel");
+        _packageVersionLabel = _stringResource.GetLocalized("WSLPackageVersionLabel");
+        _curState = _wslManager.IsDistributionRunning(Id) ? ComputeSystemState.Running : ComputeSystemState.Stopped;
+    }
+
+    private void OnStateSyncRequested(object? sender, HashSet<string> runningDistributions)
+    {
+        var newState = runningDistributions.Contains(Id)
+            ? ComputeSystemState.Running
+            : ComputeSystemState.Stopped;
+
+        UpdateState(newState);
+    }
+
+    private void UpdateState(ComputeSystemState newState)
+    {
+        lock (_lock)
+        {
+            if (_curState != newState)
+            {
+                _curState = newState;
+                StateChanged?.Invoke(this, newState);
+            }
+        }
+    }
+
+    private ComputeSystemState GetState()
+    {
+        lock (_lock)
+        {
+            return _curState;
+        }
+    }
+
+    public void RemoveSubscriptions()
+    {
+        _wslManager.DistributionStateSyncEventHandler -= OnStateSyncRequested;
+    }
+
+    public IAsyncOperation<ComputeSystemStateResult> GetStateAsync()
+    {
+        return Task.Run(() =>
+        {
+            try
+            {
+                if (GetState() == ComputeSystemState.Running)

```

Comment:
Can `GetState` return more than 2 values?  If it returns either `Running` or `Stopped` this can be changed to `return GetState()`

Reply:
updated, thanks